### ISSUE TITLE
If user is not logged in we get NPE

### ISSUE THREAD

**Issue by Nataniel-Antosik** [Android-Guild/Picover#170]
[Link to comment](https://github.com/intive/Picover/pull/157#discussion_r1313816378)

**Comment by MarekMacko** [comment]
@kmaslowiec Please drive the topic, since it's an outcome of your comment

**Comment by kmaslowiec** [comment]
> @kmaslowiec Please drive the topic, since it's an outcome of your comment

On it.

**Comment by kmaslowiec** [comment]
https://firebase.google.com/docs/auth/android/manage-users#get_the_currently_signed-in_user

**Comment by MarekMacko** [comment]
> https://firebase.google.com/docs/auth/android/manage-users#get_the_currently_signed-in_user

Please mention rest of the team, explain what and why you want to change. We should close it without spending time during our weekly meeting.

**Comment by kmaslowiec** [comment]
> > https://firebase.google.com/docs/auth/android/manage-users#get_the_currently_signed-in_user
> 
> Please mention rest of the team, explain what and why you want to change. We should close it without spending time during our weekly meeting.

@michal-kucznerowicz @marcinWisnia @BorysChajdas Previously, if the user was not logged in, we got NPE, the fix prevented that from happening. The CheckNotNull gives more information on why it happens and that is why I replaced !! with it. I still have an issue with the test. It goes to an infinity loop when it runs at "set display name WHEN updateUserName called". It runs but never goes to complete. No error appears. That is why PR is a draft.
